### PR TITLE
docs: add Vite playground

### PR DESCRIPTION
**Pull request by wobsoriano**
## ☕️ Reasoning

Adds a Vite + NextAuth.js integration playground.

Inspired by https://github.com/s-kris/vite-ssr-next-auth

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

### CODE REVIEW

File: apps/playground-vite/package.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "server:prod": "cross-env NODE_ENV=production tsx ./server"
+  },
+  "dependencies": {
+    "@vitejs/plugin-react": "^2.2.0",
+    "body-parser": "^1.20.1",
+    "cookie-parser": "^1.4.6",
+    "cross-env": "^7.0.3",
+    "dotenv": "^10.0.0",
+    "express": "^4.18.1",
+    "next-auth": "^4.16.4",
```
Comment (balazsorban44) [inline apps/playground-vite/package.json:16]:
```suggestion
    "next-auth": "worskspace:*",
```

Might need to rerun `pnpm i`

File: apps/playground-vite/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "node-fetch-native": "^0.1.8",
```
Comment (balazsorban44) [inline apps/playground-vite/package.json:17]: could we use `undici` it's shipped with Node 18+ and the most spec compliant AFAIK

File: apps/playground-vite/renderer/_error.page.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
```
Comment (balazsorban44) [inline apps/playground-vite/renderer/_error.page.tsx:1]: I believe the React import is unnecessary if it's not referenced. 
```suggestion
```

File: apps/playground-vite/server/index.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+// eslint-disable-next-line n/no-path-concat
+const root = `${__dirname}/..`
+
+startServer()
+
+async function startServer () {
+  const app = express()
+  app.use(bodyParser.urlencoded({ extended: false }))
+  app.use(bodyParser.json())
+  app.use(cookieParser())
```
[the remark this answers is not in the document]
Comment (balazsorban44) [reply apps/playground-vite/server/index.ts:23]: Not sure if this is false alarm (we do add CSRF protection in next-auth, but could we verify?

**Review by balazsorban44 — COMMENTED** [review]
Left some comments

File: apps/playground-vite/package.json (hunk 2 above)
Comment (wobsoriano) [reply apps/playground-vite/package.json:17]: Having problems with `@babel/cli` with Vite 3 currently

```bash
@babel/cli doesn't appear to be written in CJS, but also doesn't appear to be a valid ES module (i.e. it doesn't have "type": "module" or an .mjs extension for the entry point). Please contact the package author to fix.
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

ReferenceError: exports is not defined
    at eval (/@fs/Users/me/next-auth/packages/next-auth/next/index.js:5:23)
```

Comment (wobsoriano) [reply apps/playground-vite/package.json:17]: or let's not make it part of the workspace for now and just pin the next-auth version

**Comment by ndom91** [comment]
@wobsoriano hey I know this is kinda old, but I'm going through PRs and came across this - do you still want to get this "vanilla" react + vite playground in?

**Comment by wobsoriano** [comment]
Hey @ndom91 there's a lot going on in this PR and `vite-plugin-ssr` is now [`vike`](https://vike.dev/) so there are going to be a lot of breaking changes I need to update.

I can probably update this PR to use Vike or we can close this one and I can open a new one anytime 👊🏼

**Comment by ndom91** [comment]
@wobsoriano yeah i agree it might not be worth it trying to bend this PR back into shape. Mind if i close it? Obviously feel free to open a new updated one with vike if you want 👍

**Comment by wobsoriano** [comment]
Yes @ndom91, feel free to close it. Thankz!

**Comment by brillout** [comment]
👍 @wobsoriano Let us know if you need antyhing from our side to make this happen.

**Comment by wobsoriano** [comment]
@brillout will do! This is defo my time to learn Vike :)

**Comment by brillout** [comment]
I think you'll like the new so-called "[V1 design](https://vike.dev/migration/v1-design)" :) E.g. we're developing `vike-react-query`, `vike-react-zustand`, `vike-react-telefunc`, and a lot more is coming.

**Pull request closed without merging**